lstchain/reco/r0_to_dl1.py

```diff
@@ -267,6 +267,45 @@ def get_dl1(
     return dl1_container
 
 
+def apply_lh_fit(
+        event,
+        dl1_container,
+        fitter
+):
+    """
+    Prepare and performs the extraction of DL1 parameters using a likelihood
+    based reconstruction method.
+
+    Parameters
+    ----------
+    event: ctapipe event container
+    dl1_container: DL1ParametersContainer
+    fitter: TimeWaveformFitter
+
+    Returns
+    -------
+    DL1LikelihoodParametersContainer
+
+    """
+    # Applied to all simulated event and all physic events
+    if dl1_container.mc_type > -1000 or (event.trigger.event_type == EventType.SUBARRAY):
+        # Don't fit if the cleaning used in the seed parametrisation didn't select any pixels
+        if dl1_container.n_pixels <= 0:
+            lhfit_container = DL1LikelihoodParametersContainer(lhfit_call_status=0)
+        else:
+            try:
+                fitter(event=event, dl1_container=dl1_container)
```

Comment:
This should just be `lhfit_container = fitter(event=event, dl1_container=dl1_container)`. I.e. `__call__` should perform the fit and return the result.